clustering/infinispan/WFLY-20918_Add_max-idle_based_eviction_configuration_for_distributable_HttpSessions_SFSBs_Timers.adoc

```diff
@@ -0,0 +1,196 @@
+1. Implement idle time-based eviction for distributable HttpSessions
+2. Implement idle time-based eviction for SFSBs
+3. Implement idle time-based eviction for EJB Timers
+4. (optionally/future work) Unified monitoring and management capabilities
+
+Each phase can be potentially delivered independently, simplifying integration to the codebase if need be.
+Nevertheless, the current plan is to deliver all of these withing a single change request.
+
+== Admin Clients
+
+This feature will be fully compatible with existing admin clients:
+
+* CLI will be able to read and write the new 'idle-threshold' configuration attributes
+* HAL (Admin Console) will require updates to expose the new configuration options in the UI
+
+Follow-up issues will be created to track HAL enhancements once the core management model changes are defined.
+
+== Security Considerations
+
+This feature has negligible security impact.
+The ability to configure eviction settings is already protected by the existing management RBAC policies.
+No new security vulnerabilities are introduced, since eviction is a resource management mechanism that does not expose
+or modify application data in any way.
+Data is written to the disk same way as the existing passivation mechanism.
+
+== Test Plan
+
+This feature targets the default stability level and requires comprehensive testing:
+
+* Unit tests for eviction logic verifying that objects are evicted after the configured idle time
+* Integration tests verifying correct behavior when both count-based and idle-based eviction are configured
+* Integration tests for each distributed state type (HttpSessions, SFSBs, Timers)
+* Performance tests to ensure eviction processing does not introduce significant overhead
+* Manual testing of edge cases: rapid access patterns preventing eviction, configuration changes at runtime, eviction under load
+* Compatibility testing with existing configurations to ensure backwards compatibility
+* Management model tests verifying the new attributes are properly exposed via CLI and management API
+
+Testing will verify:
+
+* Objects idle longer than the configured max-idle time are evicted
+* Recently accessed objects are not evicted regardless of overall cache size
+* Eviction statistics are accurately reported
+* Configuration changes take effect without requiring server restart
+* Behavior under high concurrency and load conditions
+
+== Community Documentation
+
+Documentation will be added to the WildFly documentation as part of the implementation PR.
+Specific sections will cover:
+
+* Configuration of max-idle based eviction for each distributed state type
+* Interaction between count-based and idle-based eviction
+* Best practices for choosing between eviction strategies
+* Monitoring and troubleshooting eviction behavior
+
+== Release Note Content
+
+WildFly now supports idle time-based eviction (using the `idle-thresold` attributes) for distributable HttpSessions, Stateful JEB Beans, and JEB Timers.
```

Comment:
"EJB"

Reply:
I will change this back, looks like JEB didn't really catch up :)